portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,14 +53,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get response from the form
     String text = request.getParameter("text-input");
-    long timestamp = System.currentTimeMillis();
     if (text != null && !text.isEmpty()) {
-      myComments.add(text);
+        long timestamp = System.currentTimeMillis();
+        Entity comment = new Entity("Comment");
+        comment.setProperty("text", text);
+        comment.setProperty("timestamp", timestamp);
```

Comment:
Pull these three strings into a constant in your class and use it here and also in `doGet`. This way you don't run into typo problems.

```java
public class DataServlet extends HttpServlet {
  private static final String COMMENT_TABLE_NAME = "Comment";
  private static final String COMMENT_COLUMN_NAME = "text";
  private static final String TIMESTAMP_COLUMN_NAME = "timestamp";
```
Also note that a column name `timestamp` is not very useful. It's better if the column name describes what is the value represents and not just the type name. For example in your case this is `submit_time` or `publish_time`.

Reply:
sounds good! will incorporate this into my code!